ocaml/typing/oprint.ml

```diff
@@ -445,55 +360,43 @@ let string_of_gbl_space = function
   | Ogf_global -> "global_ "
   | Ogf_unrestricted -> ""
 
-let rec print_out_type_0 mode ppf =
+let rec print_out_type_0 ppf =
   function
   | Otyp_alias {non_gen; aliased; alias } ->
     fprintf ppf "@[%a@ as %a@]"
-      (print_out_type_0 mode) aliased
+      print_out_type_0 aliased
       (ty_var ~non_gen) alias
   | Otyp_poly ([], ty) ->
-      print_out_type_0 mode ppf ty  (* no "." if there are no vars *)
+      print_out_type_0 ppf ty  (* no "." if there are no vars *)
   | Otyp_poly (sl, ty) ->
       fprintf ppf "@[<hov 2>%a.@ %a@]"
         pr_var_jkinds sl
-        (print_out_type_0 mode) ty
+        print_out_type_0 ty
   | ty ->
-      print_out_type_1 mode ppf ty
+      print_out_type_1 ppf ty
 
 (* We must parenthesize a labeled tuple with the first element labeled when:
    - It is an argument to a function ([~arg])
    - Or, there is at least one mode to print.
  *)
 and print_out_type_mode ~arg mode ppf ty =
-  let is_local = is_local mode in
-  let is_unique = is_unique mode in
-  let is_once = is_once mode in
+  let has_modes =
+    match mode with
+    | [] -> false
+    | _ -> true
+  in
```

Comment:
```suggestion
  let has_modes = not (List.is_empty mode) in
```

Reply:
For some reason `List.is_empty` is not available here - do you know why?

Reply:
Blargh. It's in 5.1. I actually tried to upgrade my system compiler to 5.1.1 just for this, but it ran into trouble, and I have no appetite for solving the trouble. Just keep what you have.

Reply:
Yeah I realised this last night. I also tried to update my system compiler to 5 but ran into trouble.